core/src/commonMain/kotlin/com/xebia/functional/xef/auto/Conversation.kt

```diff
@@ -101,10 +158,37 @@ constructor(
    * @param numberImages number of images to generate.
    * @param size the size of the images to generate.
    */
+  @AiDsl
+  @JvmSynthetic
   suspend fun Images.images(
     prompt: Prompt,
     numberImages: Int = 1,
     size: String = "1024x1024",
     promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
   ): ImagesGenerationResponse = images(prompt, store, numberImages, size, promptConfiguration)
+
+  fun CoroutineScope.imagesAsync(
+    images: Images,
+    prompt: Prompt,
+    numberImages: Int = 1,
+    size: String = "1024x1024",
+    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
+  ): Deferred<ImagesGenerationResponse> = async {
+    images.images(prompt, store, numberImages, size, promptConfiguration)
+  }
+
+  companion object {
+
+    operator fun invoke(
+      store: VectorStore,
+      conversationId: ConversationId? = ConversationId(UUID.generateUUID().toString())
+    ): PlatformConversation = PlatformConversation.create(store, conversationId)
```

Comment:
Having `PlatformConversation` as `expect` allows us to define factories and constructors for it in the common module.